python/cudf/cudf/core/dataframe.py

```diff
@@ -1123,7 +1123,15 @@ def __setitem__(self, arg, value):
                     for col_name in self._data:
                         self._data[col_name][mask] = value
             else:
-                if isinstance(value, DataFrame):
+                if isinstance(value, (cupy.ndarray, np.ndarray)):
+                    _setitem_with_dataframe(
```

Comment:
I see the array was being converted to dataframe in order to make use of `_setitem_with_dataframe`. Ideally I wish we can separate the logic of column selection from this helper and directly replace the columns targeted. But that's out of the scope for sure.

Reply:
Yeah I agree, it would be nice to just convert the array to a column without having to deal with all the DataFrame junk but I don't see an easy way to do that at present :(